pkg/client/client.go

```diff
@@ -1674,6 +1674,17 @@ func (c *Client) PodCapacity(ctx context.Context) (int, error) {
 	return int(podCapacityTotal), nil
 }
 
+func (c *Client) GetCurrentTelemeterSecretData(ctx context.Context, s *v1.Secret) (map[string][]byte, error) {
```

Comment:
there's nothing really specific to the telemeter secret here. I'd advise to either have a more generic name and return a *v1.Secret object or use the existing GetSecret() method and move the `apierrors.IsNotFound` test to the task.

Reply:
Yes, it makes sense, after the reviews, it changed a lot. I've ended up just using `GetSecret`